articles/how-to-consume-data-from-an-API-using-react/index.md

```diff
@@ -0,0 +1,424 @@
+
+We can handle server and network errors using a `try/catch` block. We send a network request inside the `try` block . If an error occurs, the catch block will be executed.
+
+Here is the code for the `getMovie` function.
+
+```JSX
+const getMovie = async () => {
+  try {
+     const response = await fetch(url);
+      const data = await response.json()
+      setMovie(data);
+   } catch (e) {
+       console.error(e.toString);
+   } 
+}
+```
+
+This method is also executed when the component is rendered for the first time using the `useEffect` hook to load some initial data.
+
+```JSX
+useEffect(()=> {
+   getMovie(); //fetch data from api
+}, []);
+```
+
+#### 2.	onInputChange
+
+This method helps in updating the search query state when the value of the input changes, The value of the search state is updated using the `setSearch` method.
+
+```JSX
+const onInputChange = e => {
+   setSearch(e.target.value);
+}
+```
+
+### Step 5: Building the UI
+
+After we fetch the data, we should display it to the user. In the `App.js` file, we should include an input field and a button for search. The page also has an image, title, and paragraph tags. Here is the code for our app's layout.
+
+```JSX
+return(
+   <div>
+     <input type="text" value={search} onChange={onInputChange}/>
+     <button type="submit" onClick={getMovie}>Search</button>
+     
+     <img src={movie.Poster} alt=""/>
+     <h4>Title: {movie.Title}</h4>
+     <p>Year: {movie.Year}</p>
+     <p>Writer: {movie.Writer}</p>
+   </div>
+);
+```
+
+As shown above, the `getMovie` method is called whenever the button is clicked. 
+
+The `onInputChange` function is called when the value of the input changes.
+
+Here is the complete code for the `App.js` component
+
+```JSX
+import './App.css';
+import React, {useEffect, useState} from 'react';
+
+function App() {
+    const [movie, setMovie] = useState([]);
+    const [search, setSearch] = useState('');
+    const API_KEY = "cebd9b53";
+    const url = `http://www.omdbapi.com/?t=${search}&apikey=${API_KEY}`;
+
+    const getMovie = async()=>{
+      try {
+        const response = await fetch(url);
+         const data = await response.json()
+         setMovie(data);
+      } catch (e) {
+          console.error(e);
+      } 
+    }
+
+    const onInputChange = e =>{
+      setSearch(e.target.value);
+    }
+
+    useEffect(()=> {
+      getMovie();
+    }, []);
+
+    return(
+      <div>
+        <input type="text" value={search} onChange={onInputChange}/>
+        <button type="submit" onClick={getMovie}>Search</button>
+         <br></br>
+         if(movie==null){
+           <p>Movie Not Found</p>
+         }else{
+            <img src={movie.Poster} alt=""/>
+            <h4>Title: {movie.Title}</h4>
+            <p>Year: {movie.Year}</p>
+            <p>Writer: {movie.Writer}</p>
+         }
+      </div>
+    );
```

Comment:
You can't have an if condition inside JSX. This is throwing an error.

Reply:
I cannot see this code. I have this. If else is not inside jsx
function checkResponse(data){
   if(data.Response==="True"){
    return(
      <div>
         <!-- <img src={data.Poster} alt=""/>
         <h4>Title: {data.Title}</h4>
         <p>Year: {data.Year}</p>
         <p>Writer: {data.Writer}</p>
         <p>{data.Response}</p> -->
      </div>
    );
   }

   return (
     <p>No Movie found</p>
   );
 }

Reply:
@WanjaMIKE check the line number to find the code.

It's in Step 5. The final code for App.js that you have provided.